### PR TITLE
add filename for relateddependencies to xml/json reports

### PR DESCRIPTION
**Pull request by valentijnscholten**
## Fixes Issue #
Implements item 3 from #2608: add filename in xml/json reports for related dependencies

## Have test cases been added to cover the new functionality?
testcases have been updated

### CODE REVIEW

**Review by jeremylong — CHANGES_REQUESTED** [review]
Thanks for the PR! While a very limited number of JAR files are included as test resources - the preference is to add them as optional test dependencies in the `pom.xml`.  Mind updating this to remove the JAR file and include it via the pom?

File: core/src/main/resources/schema/dependency-check.2.4.xsd [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+            <xs:element name="id" type="xs:string" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="url" type="xs:string" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="description" type="xs:string" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="notes" type="xs:string" minOccurs="0" maxOccurs="1"/>
+        </xs:sequence>
+        <xs:attribute name="confidence" type="xs:string" use="optional"/>
+    </xs:complexType>
+    <xs:complexType name="relatedDependency">
+        <xs:sequence>
+            <xs:element name="fileName" type="xs:string" minOccurs="1" maxOccurs="1"/>
```
Comment (jeremylong) [inline core/src/main/resources/schema/dependency-check.2.4.xsd:63]: To make the schema backwards compatible I think this should be an optional (e.g. minOccurs="0":

```suggestion
            <xs:element name="fileName" type="xs:string" minOccurs="0" maxOccurs="1"/>
```

Comment (valentijnscholten) [reply core/src/main/resources/schema/dependency-check.2.4.xsd:63]: Isn't the idea that xml files have an xsd declaration in them that points to the schema version? If we need minOccurs=0, we could never add a "mandatory" field somewhere?

**Comment by valentijnscholten** [comment]
> Thanks for the PR! While a very limited number of JAR files are included as test resources - the preference is to add them as optional test dependencies in the `pom.xml`. Mind updating this to remove the JAR file and include it via the pom?

I could see if there is an example already that does this? In the PR I used the existing mechanism in that test, but it feels a bit weird to includ jars indeed.

**Comment by jeremylong** [comment]
@valentijnscholten Most of the test dependencies are defined in the core project pom.xml:

https://github.com/jeremylong/DependencyCheck/blob/73df62ee0e1407d99f1ef38b52785feef3d6ca98/core/pom.xml#L426-L432

https://github.com/jeremylong/DependencyCheck/blob/a503ca0eb95a7162a13fd1ff239706978909275a/core/src/test/java/org/owasp/dependencycheck/analyzer/JarAnalyzerTest.java#L174-L182

There are several JARs in the resources directory that are also used by `JarAnalyzerTest.java` - however, in some of these cases the JAR was slightly modified or does not exist in Central.

**Comment by valentijnscholten** [comment]
Commit:
- use maven for test dependency 
- use v2.4 everywhere for generating report and in targetnamespace

Would you let me know if you want that `minOccurs` above to be 0 or 1?

File: core/src/main/resources/schema/dependency-check.2.4.xsd (hunk 1 above)
Comment (jeremylong) [reply core/src/main/resources/schema/dependency-check.2.4.xsd:63]: As we are likely going to make the next version a major release (i.e. breaking changes) I'm okay with leaving this as 1 now.

**Review by jeremylong — APPROVED** [review]

**Comment by valentijnscholten** [comment]
Not sure if a milestone label is needed, but I believe this will be in the upcoming 6.0.0

**Pull request merged**